pallets/messages/src/lib.rs

```diff
@@ -51,7 +53,7 @@ pub mod pallet {
 
 		/// The maximum number of messages in a block
 		#[pallet::constant]
-		type MaxMessagesPerBlock: Get<u16>;
+		type MaxMessagesPerBlock: Get<u32>;
```

Comment:
What motivated this decision?

Reply:
So the reason is that `Decode` Trait is only implemented for Get<T : u32> when used in `BoundedVec`. So if you are using `Get<16>` to define the max bound for any storage it complains that `Decode` is not implemented for the storage.

https://github.com/paritytech/substrate/blob/polkadot-v0.9.18/frame/support/src/storage/bounded_vec.rs#L89